pyvista/core/_validation/check.py

```diff
@@ -10,36 +10,69 @@
 
 """
 
-from collections.abc import Iterable, Sequence
-from numbers import Number
-from typing import Tuple, Union, get_args, get_origin
+import math
+from numbers import Number, Real
+import operator
+import reprlib
+from typing import (
+    Any,
+    Iterable,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    Union,
+    cast,
+    get_args,
+    get_origin,
+)
 
 import numpy as np
 
-from pyvista.core._validation._cast_array import _cast_to_numpy
+from pyvista.core._typing_core import NumpyArray, Vector
+from pyvista.core._typing_core._array_like import _ArrayLike, _ArrayLikeOrScalar, _NumberType
+from pyvista.core._validation._array_wrapper import (
+    _ArrayLikeWrapper,
+    _NestedSequenceWrapper,
+    _SequenceWrapper,
+)
 
+_ScalarShape = Tuple[()]
+_ArrayShape = Tuple[int, ...]
+_Shape = Union[_ScalarShape, _ArrayShape]
+_ShapeLike = Union[int, _Shape]
+_DTypeLike = Union[np.dtype, Type[Any], str]  # type: ignore[type-arg]
 
-def check_subdtype(arg1, arg2, /, *, name='Input'):
-    """Check if a data-type is a subtype of another data-type(s).
+
+def check_subdtype(
+    input_obj: Union[_DTypeLike, _ArrayLikeOrScalar[_NumberType]],
+    /,
+    base_dtype: Union[_DTypeLike, Tuple[_DTypeLike, ...], List[_DTypeLike]],
+    *,
+    name: str = 'Input',
+):
+    """Check if an input's data-type is a subtype of another data-type(s).
 
     Parameters
     ----------
-    arg1 : dtype_like | array_like
-        ``dtype`` or object coercible to one. If ``array_like``, the dtype
-        of the array is used.
+    input_obj : DType | Array | Scalar
```

Comment:
The docstring arg types for arrays, scalars, dtypes, shapes, etc. of all `validate` and `check` functions need to be reviewed and made consistent. Since the function signatures are annotated with actual type hints, the docstring types don't necessarily need to be exact, e.g. can use `ArrayLike`, `array-like`, or `Array[float]`... I'm not sure what the convention should be.

Either way, help with reviewing these to make them fairly consistent across the package would be appreciated.

Reply:
> The docstring arg types for arrays, scalars, dtypes, shapes, etc. of all validate and check functions need to be reviewed and made consistent. Since the function signatures are annotated with actual type hints, the docstring types don't necessarily need to be exact, e.g. can use ArrayLike, array-like, or Array[float]... I'm not sure what the convention should be.

That's something I've struggled with before. Really it's kinda silly to have to duplicate the types from the signature to the docstr, since the signature is actually going to be checked by mypy. I wish numpydoc could just do it automatically.

Reply:
With #5621 I'm thinking that the annotations and docstrings should match, e.g. `Array[float]` in both annotation and docstrings. Since #5621 makes the aliases linkable, at least this way it will be clear.

I can see use cases where we may not necessarily want these to match, e.g. the annotations are complex but we want to simplify the docstring for readability. 

While working on #5285 I also found that `pandas` does this somewhat, where their docs would say `npt.NDArray[float]` but the annotations would say `npt.NDArray[np.float64]` since mypy would complain otherwise about using `float` for a numpy array (which is exactly the issue that #5285 fixed for us here).

I believe there may also be options within sphinx to display types in annotations-only, docstrings-only, or both. Not sure though

Reply:
This issue can be dealt with separately as part of #5665 I think